### PR TITLE
[FLINK-18198][docs-zh]Translate "HBase SQL Connector" page into Chinese

### PR DESCRIPTION
**Pull request by bigdata-ny**
## What is the purpose of the change

Translate "HBase SQL Connector" page into Chinese

The page url is https://ci.apache.org/projects/flink/flink-docs-master/zh/dev/table/connectors/hbase.html

The markdown file is located in flink/docs/dev/table/connectors/hbase.zh.md

## Brief change log

translate  flink/docs/dev/table/connectors/hbase.zh.md

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @bigdata-ny , there is some updates in `hbase.zh.md`. Could you update the corresponding translation ? I will review it then.

**Comment by bigdata-ny** [comment]
> Hi @bigdata-ny , there is some updates in `hbase.zh.md`. Could you update the corresponding translation ? I will review it then.

@wuchong done.

File: docs/dev/table/connectors/hbase.zh.md [hunk 1]
```diff
@@ -31,29 +31,29 @@ under the License.
 * This will be replaced by the TOC
 {:toc}
 
-The HBase connector allows for reading from and writing to an HBase cluster. This document describes how to setup the HBase Connector to run SQL queries against HBase.
+HBase 连接器支持读取和写入 HBase 集群。本文档介绍如何使用 HBase 连接器基于 HBase 进行 SQL 查询。
 
-HBase always works in upsert mode for exchange changelog messages with the external system using a primary key defined on the DDL. The primary key must be defined on the HBase rowkey field (rowkey field must be declared). If the PRIMARY KEY clause is not declared, the HBase connector will take rowkey as the primary key by default.
+HBase 连接器在 upsert 模式下运行，可以使用 DDL 中定义的一个主键与外部系统交换更新操作消息。但是主键只能基于 HBase 的 rowkey 字段定义。如果没有声明主键，HBase 连接器默认取 rowkey。
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:36]:
```suggestion
HBase 连接器在 upsert 模式下运行，可以使用 DDL 中定义的主键与外部系统交换更新操作消息。但是主键只能基于 HBase 的 rowkey 字段定义。如果没有声明主键，HBase 连接器默认取 rowkey 作为主键。
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 2]
```diff
@@ -31,29 +31,29 @@ under the License.
...
 
-Dependencies
+依赖
 ------------
 
-In order to setup the HBase connector, the following table provide dependency information for both projects using a build automation tool (such as Maven or SBT) and SQL Client with SQL JAR bundles.
+安装 HBase 连接器的依赖条件如下表，包括自动构建工具（比如 Maven 或者 SBT）和 SQL 客户端：
 
-| HBase Version       | Maven dependency                                          | SQL Client JAR         |
+| HBase 版本          | Maven 依赖                                          | SQL 客户端 jar        |
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:43]:
```suggestion
| HBase 版本          | Maven 依赖                                          | SQL 客户端 JAR        |
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 3]
```diff
@@ -31,29 +31,29 @@ under the License.
...
+| 1.4.x               | `flink-connector-hbase{{site.scala_version_suffix}}`      | {% if site.is_stable %} [Download](https://repo.maven.apache.org/maven2/org/apache/flink/flink-connector-hbase{{site.scala_version_suffix}}/{{site.version}}/flink-connector-hbase{{site.scala_version_suffix}}-{{site.version}}.jar) {% else %} 只适用于 [稳定发布版]({{ site.stable_baseurl }}/zh/dev/table/connectors/hbase.html) {% endif %}|
 
 
-How to create an HBase table
+如何使用 HBase 表
 ----------------
 
-All the column families in HBase table must be declared as ROW type, the field name maps to the column family name, and the nested field names map to the column qualifier names. There is no need to declare all the families and qualifiers in the schema, users can declare what’s used in the query. Except the ROW type fields, the single atomic type field (e.g. STRING, BIGINT) will be recognized as HBase rowkey. The rowkey field can be arbitrary name, but should be quoted using backticks if it is a reserved keyword.
+所有 HBase 表的列簇必须定义为 ROW 类型，字段名对应列簇名，嵌套的字段名对应列名。用户只需在表结构中声明查询中使用的的列簇和列。除了 ROW 类型的列，其它原子数据类型（比如，字符串、bigint）将被识别为 HBase 的 rowkey。rowkey 字段必须是唯一的，如果是保留关键字，需要用反引号。
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:51]:
```suggestion
所有 HBase 表的列簇必须定义为 ROW 类型，字段名对应列簇名（column family），嵌套的字段名对应列限定符名（column qualifier）。用户只需在表结构中声明查询中使用的的列簇和列限定符。除了 ROW 类型的列，剩下的原子数据类型字段（比如，STRING, BIGINT）将被识别为 HBase 的 rowkey，一张表中只能声明一个 rowkey。rowkey 字段的名字可以是任意的，如果是保留关键字，需要用反引号。
```

column qualifier 是翻译成 "列限定符" 吧？如果没有标准翻译，建议不翻译。

File: docs/dev/table/connectors/hbase.zh.md [hunk 4]
```diff
@@ -66,123 +66,116 @@ CREATE TABLE hTable (
...
+-- 假设 "T" 的表结构是 [rowkey, f1q1, f2q2, f2q3, f3q4, f3q5, f3q6]
 INSERT INTO hTable
 SELECT rowkey, ROW(f1q1), ROW(f2q2, f2q3), ROW(f3q4, f3q5, f3q6) FROM T;
 
--- scan data from the HBase table
+-- 从 HBase 表扫描数据
 SELECT rowkey, family1, family3.q4, family3.q6 FROM hTable;
 
--- temporal join the HBase table as a dimension table
+-- temporal join HBase 表为一个维度表
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:77]:
```suggestion
-- temporal join HBase 表，将 HBase 表作为维表
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 5]
```diff
@@ -66,123 +66,116 @@ CREATE TABLE hTable (
...
     <tbody>
     <tr>
       <td><h5>connector</h5></td>
-      <td>required</td>
-      <td style="word-wrap: break-word;">(none)</td>
+      <td>必选</td>
+      <td style="word-wrap: break-word;">none</td>
       <td>String</td>
-      <td>Specify what connector to use, here should be <code>'hbase-1.4'</code>.</td>
+      <td>指定使用的连接器，这里写“hbase-1.4”</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:104]: 末位增加句号。下面也一样。

File: docs/dev/table/connectors/hbase.zh.md [hunk 6]
```diff
@@ -66,123 +66,116 @@ CREATE TABLE hTable (
...
       <td><h5>sink.buffer-flush.interval</h5></td>
-      <td>optional</td>
+      <td>可选</td>
       <td style="word-wrap: break-word;">1s</td>
       <td>Duration</td>
-      <td>Writing option, the interval to flush any buffered rows.
-      This can improve performance for writing data to HBase database, but may increase the latency.
-      Can be set to <code>'0'</code> to disable it. Note, both <code>'sink.buffer-flush.max-size'</code> and <code>'sink.buffer-flush.max-rows'</code>
-      can be set to <code>'0'</code> with the flush interval set allowing for complete async processing of buffered actions.
+      <td>写入的参数选项。刷写缓存行的间隔。无默认值意味着没有异步刷写线程将被调度。例如，“1s”， “5 s”
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:155]: 这段后来有更新，请重新翻译下。

File: docs/dev/table/connectors/hbase.zh.md [hunk 7]
```diff
@@ -66,123 +66,116 @@ CREATE TABLE hTable (
...
 ----------------
 
-HBase stores all data as byte arrays. The data needs to be serialized and deserialized during read and write operation
+HBase 以字节数组存储所有数据。在读和写过程中要序列化和反序列化数据。
 
-When serializing and de-serializing, Flink HBase connector uses utility class `org.apache.hadoop.hbase.util.Bytes` provided by HBase (Hadoop) to convert Flink Data Types to and from byte arrays.
+Flink 的 HBase 连接器利用 HBase（Hadoop) 的工具类 org.apache.hadoop.hbase.util.Bytes 进行字节数组和 Flink 数据类型转换。
 
-Flink HBase connector encodes `null` values to empty bytes, and decode empty bytes to `null` values for all data types except string type. For string type, the null literal is determined by `null-string-literal` option.
+Flink 的 HBase 连接器将所有数据类型（除字符串外）null 值编码成空字节。对于字符串类型，null 字面值由“null-string-literal”选项值决定。
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:170]:
```suggestion
Flink 的 HBase 连接器将所有数据类型（除字符串外）`null` 值编码成空字节。对于字符串类型，`null`值的字面值由`null-string-literal`选项值决定。
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 8]
```diff
@@ -273,29 +266,29 @@ double toDouble(byte[] bytes)
     </tr>
     <tr>
       <td><code>DATE</code></td>
-      <td>Stores the number of days since epoch as int value.</td>
+      <td>从 1970-01-01 00:00:00 UTC 开始的天数，整数型</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:269]:
```suggestion
      <td>从 1970-01-01 00:00:00 UTC 开始的天数，int 值。</td>
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 9]
```diff
@@ -273,29 +266,29 @@ double toDouble(byte[] bytes)
...
     </tr>
     <tr>
       <td><code>TIME</code></td>
-      <td>Stores the number of milliseconds of the day as int value.</td>
+      <td>从 1970-01-01 00:00:00 UTC 开始天的毫秒数，整数型</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:273]:
```suggestion
      <td>从 1970-01-01 00:00:00 UTC 开始天的毫秒数，int 值。</td>
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 10]
```diff
@@ -273,29 +266,29 @@ double toDouble(byte[] bytes)
...
     </tr>
     <tr>
       <td><code>TIMESTAMP</code></td>
-      <td>Stores the milliseconds since epoch as long value.</td>
+      <td>从 1970-01-01 00:00:00 UTC 开始的毫秒数，长整型</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:277]:
```suggestion
      <td>从 1970-01-01 00:00:00 UTC 开始的毫秒数，long 值。</td>
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 11]
```diff
@@ -66,123 +66,116 @@ CREATE TABLE hTable (
...
+      <td>HBase 集群的 Zookeeper 根目录</td>
     </tr>
     <tr>
       <td><h5>null-string-literal</h5></td>
-      <td>optional</td>
+      <td>可选</td>
       <td style="word-wrap: break-word;">null</td>
       <td>String</td>
-      <td>Representation for null values for string fields. HBase source and sink encodes/decodes empty bytes as null values for all types except string type.</td>
+      <td>字符串为 null 时取值。HBase 的 source 和 sink 的编解码将所有数据类型（除字符串外）的空字节转为 null</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:132]:
```suggestion
      <td>当字符串值为 <code>null</code> 时的存储形式，默认存成 "null" 字符串。HBase 的 source 和 sink 的编解码将所有数据类型（除字符串外）将 <code>null</code> 值以空字节来存储。</td>
```

File: docs/dev/table/connectors/hbase.zh.md [hunk 12]
```diff
@@ -66,123 +66,115 @@ CREATE TABLE hTable (
...
+        <th class="text-center" style="width: 50%">参数描述</th>
       </tr>
     </thead>
     <tbody>
     <tr>
       <td><h5>connector</h5></td>
-      <td>required</td>
-      <td style="word-wrap: break-word;">(none)</td>
+      <td>必选</td>
+      <td style="word-wrap: break-word;">none</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/hbase.zh.md:102]: Keep the original bracket `( )`

**Pull request closed without merging**